File: dp/geeksForGeeksProblems/Easy Problems/nthCatalanNumber.cpp
#include<bits/stdc++.h>
#define ll long long
#define m 1000000007
using namespace std;
ll dp[101]={0};
void initializeDP(){
    dp[0]=1;
    for(ll i=1;i<101;i++){
        for(ll j=0;j<i;j++){
            dp[i]+=(dp[j]*dp[i-j-1]);
        }
    }
    // for(ll i=0;i<100;i++)
    //     cout<<dp[i]<<" ";
}
ll solve(int n){
    return dp[n];
}
int main(){
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    ll t;
    cin>>t;
    // cout<<"hello"<<endl;
    initializeDP();
    while(t--){
        ll n;
        cin>>n;
        cout<<solve(n)<<endl;
    }
    return 0;
}

File: dp/geeksForGeeksProblems/Easy Problems/nthCatalanNumber.py
import sys 
sys.stdin = open('input.txt', 'r')  
sys.stdout = open('output.txt', 'w') 
dp=[0 for i in range(101)]
def initialiseDP():
    dp[0]=1
    for i in range(1,101):
        for j in range(i):
            dp[i]+=(dp[j]*dp[i-j-1])
    # print(dp)
def solve(n):
    return dp[n]
t=int(input())
initialiseDP()
while t:
    n=int(input())
    print(solve(n))
    t-=1